### PR TITLE
Handle supported mechanisms in `hello` and `getParameters` commands

### LINKED ISSUE #4365: Remove `PLAIN` support for new authentication
### What should be done?

When FerretDB is started with old authentication, only `PLAIN` should be supported (as it is now).
When FerretDB is started with new authentication, only `SCRAM` authentication methods should be supported.

We should refactor our code to split `PLAIN` and `SCRAM` code paths into separate functions/methods as much as possible. That should simplify old authentication and `PLAIN` removal in 2.0.

We don't test new authentication in integration tests, so this issue should be tested manually.

### Where?

Handler and other places.

### Definition of Done

- handler updated;
- all backends updated;
- unit tests updated;
- integration/compatibility tests updated;
- refactoring done.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #4365.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by AlekSi — COMMENTED** [review]
There are conflicts now

File: internal/handler/msg_hello.go [hunk 1]
```diff
@@ -73,9 +73,15 @@ func (h *Handler) MsgHello(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, e
 		)
 	}
 
-	mechs, err := h.getUserSupportedMechs(ctx, db, username)
-	if err != nil {
-		return nil, lazyerrors.Error(err)
+	// for the backend authentication return `PLAIN` regardless of the user existence,
+	// getting backend users is not easy
```
Comment (AlekSi) [inline internal/handler/msg_hello.go:77]: Well, it is not about being "easy" – those backend users may not exist (SQLite) or be unrelated users (PostgreSQL)

**Review by AlekSi — APPROVED** [review]

**Review by noisersup — APPROVED** [review]
LGTM!

File: internal/handler/msg_hello.go (hunk 1 above)
Comment (chilagrow) [reply internal/handler/msg_hello.go:77]: Sure, let's remove that comment :+1:

**Pull request merged**